subscription_oca/models/sale_order.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2023 Domatix - Carlos Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from collections import defaultdict
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    subscription_ids = fields.One2many(
+        comodel_name="sale.subscription",
+        inverse_name="sale_order_id",
+        string="Subscriptions",
+    )
+
+    subscriptions_count = fields.Integer(compute="_compute_subscriptions_count")
+
+    order_subscription_id = fields.Many2one(
+        comodel_name="sale.subscription", string="Subscription"
+    )
+
+    @api.depends("subscription_ids")
+    def _compute_subscriptions_count(self):
+        for subscriptions in self:
+            subscriptions.subscriptions_count = len(self.subscription_ids)
+
+    def action_view_subscriptions(self):
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "sale.subscription",
+            "domain": [("id", "in", self.subscription_ids.ids)],
+            "name": self.name,
+            "view_mode": "tree,form",
+        }
+
+    def get_next_interval(self, type_interval, interval):
+        date_start = date.today()
+        if type_interval == "days":
+            date_start += relativedelta(days=+interval)
```

Comment:
May be its better to write something like this for all cases:
`
date_start += relativedelta(**{
    type_interval: interval
})
`